src/composite.c

```diff
@@ -0,0 +1,119 @@
+#include <stdio.h>
+#include <vips/vips.h>
+
+/* Composite images `src` and `dst`.
+ */
+int Composite(VipsObject *context, VipsImage *src, VipsImage *dst,
+      VipsImage **out) {
+  const int ALPHA_BAND_INDEX = 3;
+  const int NUM_COLOR_BANDS = 3;
+
+  // These images will all be unreffed when context is unreffed:
+  VipsImage **t = (VipsImage **) vips_object_local_array(context, 3);
+
+  // Extract RGB bands:
+  VipsImage *srcRGB;
+  VipsImage *dstRGB;
+  if (vips_extract_band(src, &srcRGB, 0, "n", NUM_COLOR_BANDS, NULL) ||
+      vips_extract_band(dst, &dstRGB, 0, "n", NUM_COLOR_BANDS, NULL))
+    return -1;
+
+  vips_object_local(context, srcRGB);
+  vips_object_local(context, dstRGB);
```

Comment:
Thanks for the advice with `vips_object_local`. Please check that I made that change correctly.